atlasdb-feign/src/main/java/com/palantir/atlasdb/http/PollingRefreshable.java

```diff
@@ -46,50 +49,58 @@
     private final Duration refreshInterval;
     private final ScheduledExecutorService poller;
 
-    private final Refreshable<T> refreshable = Refreshable.empty();
+    private final Function<T, U> transform;
+    private final Refreshable<U> refreshable = Refreshable.empty();
 
-    private T lastSeenValue;
+    private T lastSeenValue = null;
```

Comment:
Is this stuff threadsafe?

Reply:
In my opinion, it is since this value is used by only the refreshing thread. Also the value transformed is stored in `Refreshable` which is threadsafe and can be considered as a synchronization point between different threads.

Reply:
But you also set it in the constructor, that's what I referred to.

Reply:
@OStevan to explain here the problem of safe publication.

Reply:
After the last comment I had a discussion with @jkozlowski about thread safety and here are the notes.

- access to `lastSeenValue` inside the updating thread and using the `Refreshable` for reads/writes is thread safe (as previously discussed)
- however based on what we read, `PoolingRefreshable` in this implementation can not be considered thread safe as the `lastSeenValue` is not final or volatile which means that we do not have safe publication of the constructed object

Reply:
Ah, good catch! Yeah, forgot about the constructor